documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+# SDK Breaking Change and Diagnostic Guidelines
+
+This document provides guidelines for introducing new diagnostics or breaking changes to the .NET SDK, which configuration knobs are available, and what criteria should guide the decision around severity, timeline for introduction, and deprecation of diagnostics.
+
+## Overall procedure
+
+In general, we want to make updating the .NET SDK as smooth as possible for developers. This means:
+
+* introducing new changes in a staged/gradual way
+* trying to tie opinionated analyzers/diagnostics to a mechanism that requires explicit user opt-in
+* providing a way to opt out of a change entirely
+
+## Kinds of changes
+
+There are many kinds of breaking changes that can ship in the .NET SDK, such as:
+
+* New MSBuild warnings and errors (props/targets)
+* New NuGet warnings and errors. 
+  * For example, NuGet Audit.
+* Roslyn Analyzers and CodeFixes
+  * This includes trimming/ILLink analyzers and codefixes
+* Behavioral/implementation changes
+  * MSBuild engine changes like MSBuild Server
+  * Implementation changes for MSBuild Tasks
+  * NuGet Restore algorithm enhancements
+  * Changes to DotNet CLI grammar
+  * Changes to defaults in CLI flags that impact behavior
```

Comment:
changes to CLI behavior itself. For example, we're considering changing dotnet clean to clean everything, not just your tfm/rid config which is what it does today.

Reply:
Added.

Reply:
Added.